internal/identity/manager/schedulers.go

```diff
@@ -0,0 +1,161 @@
+package manager
+
+import (
+	"context"
+	"sync/atomic"
+	"time"
+
+	"github.com/pomerium/pomerium/pkg/grpc/session"
+)
+
+type updateUserInfoScheduler struct {
+	baseCtx                context.Context
+	updateUserInfoInterval time.Duration
+	updateUserInfo         func(ctx context.Context, userID string)
+	userID                 string
+
+	reset  chan struct{}
+	cancel context.CancelFunc
+}
+
+func newUpdateUserInfoScheduler(
+	ctx context.Context,
+	updateUserInfoInterval time.Duration,
+	updateUserInfo func(ctx context.Context, userID string),
+	userID string,
+) *updateUserInfoScheduler {
+	uuis := &updateUserInfoScheduler{
+		baseCtx:                ctx,
+		updateUserInfoInterval: updateUserInfoInterval,
+		updateUserInfo:         updateUserInfo,
+		userID:                 userID,
+		reset:                  make(chan struct{}, 1),
+	}
+	ctx, uuis.cancel = context.WithCancel(context.WithoutCancel(uuis.baseCtx))
+	go uuis.run(ctx)
+	return uuis
+}
+
+func (uuis *updateUserInfoScheduler) Reset() {
+	select {
+	case uuis.reset <- struct{}{}:
+	default:
+	}
```

Comment:
This effectively de-duplicates Reset() calls, is that right? Because `reset` is a buffered channel, this should never "miss" a reset, because sending to the reset channel should never block unless there's already a pending item in the channel buffer?

(not requesting any changes, just want to make sure I understand how this works)

Reply:
Added a comment. Yes we're basically using the channel as a signaling mechanism. There are other ways to accomplish this, but using channels allows us to use the `select` statement.